xbmc/cores/VideoPlayer/VideoRenderers/RenderManager.cpp

```diff
@@ -1083,9 +1085,7 @@ void CRenderManager::PrepareNextRender()
   double frameOnScreen = m_dvdClock.GetClock();
   double frametime = 1.0 / g_graphicsContext.GetFPS() * DVD_TIME_BASE;
 
-  // correct display latency
-  // internal buffers of driver, assume that driver lets us go one frame in advance
-  double totalLatency = DVD_SEC_TO_TIME(m_displayLatency) - DVD_MSEC_TO_TIME(m_videoDelay) + 2* frametime;
+  double totalLatency = DVD_SEC_TO_TIME(TotalLatency());
```

Comment:
> we should be able to calculate vblank time

wp_presentation tells us when the vblank happens for each frame we commit to the compositor with swapBuffers

> A frame becomes visible short after vblank and m_latencyMovingAverage is average time from swapBuffers to visible (at a vblank). right?

Yes, more concretely latencyMovingAverage is the average time from before eglSwapBuffers call (i.e. when PresentRender is called) to visible. So it will include the time eglSwapBuffers() blocks.

> At the time of swapBuffers ptsOnScreen should be clock - m_latencyMovingAverage. Later in PrepareNextRender the clock has advanced t(PrepNextRender) - t(swapBuffers). I would subtract this time from clock to get ptsOnScreen.

OK, that sounds like I need the time from after eglSwapBuffers(), right? I'll try that.